### DISCUSSION TITLE
LinkerdD and "hostNetwork" monitoring infrastructure

### DISCUSSION THREAD

**Discussion by zer0stars** [linkerd/linkerd2#11040] [Help]
Hello,

We have been using LinkerD for nearly a year, working great for all our applications. Two pieces of infrastructure that do not have a sidecar running today are:
- Prometheus Node Exporter
- Filebeat

How can linkerd be injected into these pods in such a way that it will not override host networking configuration, but protect outbound traffic to filbeat, and encrypt communication for prometheus scraping.

**Comment by adleong** [comment]
Hi @zer0stars.  The Linkerd proxy cannot be injected into pods that use hostNetworking.  In order to get the benefit of Linkerd for these workloads, you would need to run them in pods which don't use hostNetworking.

**Comment by Charlotte-br560** [comment]
Glad to hear LinkerD has been serving you well! To integrate LinkerD without overriding host networking configurations for Prometheus Node Exporter and Filebeat, you can utilize LinkerD's --proxy-auto-inject feature. This allows you to selectively inject the LinkerD sidecar into specific pods.

For Prometheus scraping, you can configure LinkerD to encrypt communication by adding appropriate annotations to the Prometheus deployment configuration.

As for Filebeat, ensure that outbound traffic is directed through LinkerD by adjusting routing rules or service mesh configuration accordingly.
